File: flutter/lib/features/champion/domain/usecase/get_champion_list.dart
import 'package:lol_champion/features/champion/domain/entity/champion.dart';
import 'package:lol_champion/features/champion/domain/repository/champion_repository.dart';

class GetChampionList {
  const GetChampionList(this._repository);

  final ChampionRepository _repository;

  Future<List<Champion>> call() => _repository.getChampionList();
}

File: react/src/features/champion/domain/use-cases/get-champion-list.ts
import type { Champion } from '@/features/champion/domain/entities/champion';
import type { ChampionRepository } from '@/features/champion/domain/repositories/champion-repository';

export class GetChampionList {
  constructor(private readonly repository: ChampionRepository) {}

  execute(patchVersion: string, locale: string): Promise<readonly Champion[]> {
    return this.repository.getChampionList(patchVersion, locale);
  }
}
